### PR TITLE
Pitfall section in readme.md for newcomers

### PR DESCRIPTION
**Pull request by cklam2**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngneat/until-destroy/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

There are certain pitfalls for newcomers when using this library. 

Issue Number: #91 and #109 

## What is the new behavior?
To avoid ppl searching for potential solutions in the closed issues it would be useful to have a list of tips and tricks for the most important pitfalls. New section `Potential pitfalls` in readme.md

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,6 +155,11 @@ You can use `--removeOnDestroy` flag for empty `OnDestroy` methods removing.
 npx @ngneat/until-destroy --removeOnDestroy
 ```
 
+## Potential pitfalls
```
Comment (arturovt) [inline README.md:158]:
```suggestion
## Potential Pitfalls
```

File: README.md [hunk 2]
```diff
@@ -155,6 +155,11 @@ You can use `--removeOnDestroy` flag for empty `OnDestroy` methods removing.
...
+
+- The order of decorators are important, make sure to put `@UntilDestroy()` before the `@Component` decorator.
```
Comment (arturovt) [inline README.md:160]:
```suggestion
- The order of decorators is important, make sure to put `@UntilDestroy()` before the `@Component()` decorator.
```

File: README.md [hunk 3]
```diff
@@ -155,6 +155,11 @@ You can use `--removeOnDestroy` flag for empty `OnDestroy` methods removing.
...
+- If in unit test a component is overridden with [`overrideComponent`](https://angular.io/api/core/testing/TestBed#overrideComponent), it overrides metadata and component definitions. To reapply the decorator, invoke `UntilDestroy()(YourComponent);`. See [here](https://github.com/ngneat/until-destroy/issues/91#issuecomment-626470446) for an example.
```
Comment (arturovt) [inline README.md:161]:
```suggestion
- When using [`overrideComponent`](https://angular.io/api/core/testing/TestBed#overrideComponent) in unit tests remember that it overrides metadata and component definition. Invoke `UntilDestroy()(YourComponent);` to reapply the decorator. See [here](https://github.com/ngneat/until-destroy/issues/91#issuecomment-626470446) for an example.
```

**Comment by cklam2** [comment]
Thank @arturovt, applied the suggested improvements.

**Comment by jaco-terbraak** [comment]
Thanks @NetanelBasal

**Pull request merged**